### DISCUSSION TITLE
Can I use SpiceEditor to change net label names?

### DISCUSSION THREAD

**Discussion by danieljfarrell** [nunobrum/PyLTSpice#134] [Q&A]
I have a circuit with very complicated switching requirements. I've labelled the poles and throws of my switches with net labels in LTSpice GUI. I want to import this net into PyLTSpice and iterate through all the switch combinations by simply changing net labels.

For example, here the R2 is bypassed by the "switch" labels that connect A1 to A2 - a small value resistor is used to preserve both net names (jumpers seem to be broken in LTSpice). To put R2 back in the circuit I would like to edit the net label on the right hand side of the "switch" resistor R_SWITCH: change the A2 label to NOT_CONNECTED.

<img width="1247" alt="image" src="https://github.com/nunobrum/PyLTSpice/assets/112847/f31e8c3a-2f20-4630-bc1e-948085ae40cc">

**Comment by nunobrum** [comment]
Hi Daniel,

this is something that is not currently supported. 
Just an idea. Instead of updating a label, can't you just delete the resistor? 
It's probably more speaking than updating a label.

Best Regards,
Nuno

**Reply by danieljfarrell** [reply]
Hi Nuno,

Thanks for the advice. Just wondering, would you accept this as a pull request, and if so, how would you implement it i.e. where should the logic go and what gotchas might I run in to?

**Reply by nunobrum** [reply]
Hi Daniel,

yes, I accept pull requests. The only criteria is that they are written in line with the python coding standards and don't brreak existing functions.
What you are requesting is not that complicated. You can see how Texts are managed in the QschEditor. You just need to add a function to update it. 
If you implement it at the BaseEditor level, then it will be also available to LTSpice
